pkg/skaffold/initializer/init_test.go

```diff
@@ -59,3 +61,120 @@ func TestPrintAnalyzeJSON(t *testing.T) {
 		})
 	}
 }
+
+func TestWalk(t *testing.T) {
+	emptyFile := ""
+	tests := []struct {
+		name                string
+		filesWithContents   map[string]string
+		expectedConfigs     []string
+		expectedDockerfiles []string
+		force               bool
+		shouldErr           bool
+	}{
+		{
+			name: "should return correct k8 configs and dockerfiles",
+			filesWithContents: map[string]string{
+				"config/test.yaml":  emptyFile,
+				"k8pod.yml":         emptyFile,
+				"README":            emptyFile,
+				"deploy/Dockerfile": emptyFile,
+				"Dockerfile":        emptyFile,
+			},
+			force: false,
+			expectedConfigs: []string{
+				"config/test.yaml",
+				"k8pod.yml",
+			},
+			expectedDockerfiles: []string{
+				"Dockerfile",
+				"deploy/Dockerfile",
+			},
+			shouldErr: false,
```

Comment:
We usually avoid those `shouldErr: false`

Reply:
Actually, i follow the guidelines to make tests easy to read and hence spell it out there.

Reply:
I understand